tests/conftest.py

```diff
@@ -0,0 +1,252 @@
+"""
+Module containing the grids to be used in the different tests. All fixtures in this file
+are available to the tests in this directory and its subdirectories.
+"""
+
+from typing import cast
+
+import numpy as np
+import porepy as pp
+import pytest
+import scipy.sparse as sps
+
+import pygeon as pg
+
+# ------------------------- Unit simplicial grids -------------------------
+
+param_list = [(dim, is_str) for dim in range(1, 4) for is_str in [True, False]]
+# Remove the (1, True) entry because it's the same as (1, False)
+param_list = param_list[1:]
+ids = ["1D", "2D_struct", "2D_unstruct", "3D_struct", "3D_unstruct"]
+
+
+@pytest.fixture(scope="session")
+def _unit_grids_dict() -> dict[tuple[int, bool], pg.Grid]:
+    grids = {}
+    for dim, is_str in param_list:
+        sd = pg.unit_grid(dim, 1 / (5 - dim), structured=is_str, as_mdg=False)
+        sd = cast(pg.Grid, sd)  # mypy
+
+        sd.compute_geometry()
+        grids[dim, is_str] = sd
+
+    return grids
+
+
+@pytest.fixture(scope="session", params=param_list, ids=ids)
+def unit_sd(_unit_grids_dict: dict, request: pytest.FixtureRequest) -> pg.Grid:
+    return _unit_grids_dict[request.param]
+
+
+@pytest.fixture
+def unit_sd_1d(_unit_grids_dict: dict) -> pg.Grid:
+    """
+    1D grid of the unit line segment with 4 elements
+    """
+    return _unit_grids_dict[1, False]
+
+
+@pytest.fixture
+def unit_sd_2d(_unit_grids_dict: dict) -> pg.Grid:
+    """
+    Unstructured triangle grid of the unit square
+    """
+    return _unit_grids_dict[2, False]
+
+
+@pytest.fixture
+def unit_sd_3d(_unit_grids_dict: dict) -> pg.Grid:
+    """
+    Unstructured tetrahedral grid of the unit cube
+    """
+    return _unit_grids_dict[3, False]
+
+
+# ------------------------- Unit Cartesian grids -------------------------
+
+
+@pytest.fixture(scope="session")
+def _unit_cart_dict() -> dict[int, pg.Grid]:
+    grids = {}
+    for dim in [1, 2, 3]:
+        sd = pp.CartGrid(
+            np.array([5 - dim] * dim),
+            np.array([1] * dim),
+        )
+        pg.convert_from_pp(sd)
+
+        sd.compute_geometry()
+        grids[dim] = cast(pg.Grid, sd)  # mypy
```

Comment:
same comment as before, you had reasons for making this. Which one was not interpreted correctly as pg.Grid?